src/library/Connect/LocalEnvironmentManager.cs

```diff
@@ -436,12 +436,26 @@ private void _ReportProgress(EventLevel eventLevel, string message, params objec
 
                 // because we are using dns name instead of service we have to retrieve it by splitting when needed
                 // If this ever cause issues we should consider larger refactor where we add serviceName member variable to EndpointInfo class.
-                var serviceName = endpoint.DnsName
+                var dnsNameArray = endpoint.DnsName
                     .ToUpperInvariant()
-                    .Split(".")
+                    .Split(".");
+
+                var serviceName = dnsNameArray
                     .First()
-                    .Replace("-", "_")
-                    .Replace(".", "_");
+                    .Replace("-", "_");
+
+                // when !endpoint.IsInWorkloadNamespace
+                var serviceNs = dnsNameArray.Length == 2
+                    ? dnsNameArray
+                        .Last()
+                        .Replace("-", "_")
+                    : null;
+
+                // sometimes cross talk is desired between namespaces, append those not matching workload
+                if (!string.IsNullOrWhiteSpace(serviceNs))
+                {
+                    serviceName = $"{serviceName}_{serviceNs}";

```

Comment:
I chose to do it this way as this is the way we think about the services with respect to k8s dns. So it made sense to simply append the ns here rather than mess with all the other logic setting env vars and append it elsewhere